src/main/java/com/onarandombox/MultiverseCore/commands/GameruleCommand.java

```diff
@@ -0,0 +1,87 @@
+package com.onarandombox.MultiverseCore.commands;
+
+import java.util.ArrayList;
+
+import co.aikar.commands.CommandIssuer;
+import co.aikar.commands.annotation.CommandAlias;
+import co.aikar.commands.annotation.CommandCompletion;
+import co.aikar.commands.annotation.CommandPermission;
+import co.aikar.commands.annotation.Description;
+import co.aikar.commands.annotation.Optional;
+import co.aikar.commands.annotation.Subcommand;
+import co.aikar.commands.annotation.Syntax;
+import com.onarandombox.MultiverseCore.MultiverseCore;
+import com.onarandombox.MultiverseCore.api.MVWorld;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.GameRule;
+import org.bukkit.World;
+import org.jetbrains.annotations.NotNull;
+
+@CommandAlias("mv")
+public class GameruleCommand extends MultiverseCoreCommand {
+    public GameruleCommand(@NotNull MultiverseCore plugin) {
+        super(plugin);
+    }
+
+    @Subcommand("gamerule")
+    @CommandPermission("multiverse.core.gamerule")
+    @CommandCompletion("@gamerules True|False|@range:1-10 @mvworlds|*")
+    @Syntax("<Gamerule> <Gamerule value> [World or *]")
+    @Description("Changes a gamerule in 1 world")
+    public void onGameruleCommand(CommandIssuer issuer,
+
+                                @Syntax("<Gamerule>")
+                                @Description("Gamerule to set")
+                                String gamerule,
+
+                                @Syntax("<Value>")
+                                @Description("Value of gamerule")
+                                String value,
+
+                                @Optional
+                                @Syntax("[World or *]")
+                                @Description("World to apply gamerule to, current world by default")
+                                String worldOrAll
+
+
+    ) {
+        MVWorld gameruleChangingInWorld = null;
+
+        final GameRule gameRuleObject = GameRule.getByName(gamerule);
+
+        // All worlds
+        if (worldOrAll.equals("*")) {
+
+            ArrayList<Boolean> successfulGameruleAdding = new ArrayList<Boolean>();
```

Comment:
I dont see a point in an arraylist here, you can just use a boolean. Assume all is successful and set to `true` first. Then when any of the gamerule setting fails, change it to `false`.

Reply:
The issue with that is, if one world fails then the one after it succeeds, it will think that all succeeded... And say so (unless I am missing something)

If you wanted to break after an unsuccessful gamerule setting then it would leave some set and others not, so you would have to set all gamerules → check if successful → if 1 fails then revert all, and that seems like too much work for something that would almost never be called.

Reply:
the logic is "if any is false I print to failed message", so we can just do:
```java
boolean success = true;
for (...) {
    if (action failed) {
        success = false;
        continue;
    }
    // Do wtv is needed if action passed
    // ...
}
if (!success) {
    // print error message
}
```